src/main/java/com/linkedin/kmf/services/MultiClusterTopicManagementService.java

```diff
@@ -287,14 +289,32 @@ int minPartitionNum() throws InterruptedException, ExecutionException {
       return Math.max((int) Math.ceil(_minPartitionsToBrokersRatio * brokerCount), _minPartitionNum);
     }
 
+    private List<TopicPartitionInfo> topicPartitionsWithRetry(KafkaFuture<TopicDescription> topicDescriptionKafkaFuture)
+        throws InterruptedException {
+      while (true) {
+        boolean processedSuccessfully = true;
+        long sleepTime = TimeUnit.SECONDS.toMillis(2);
+        do {
+          try {
+            return topicDescriptionKafkaFuture.get().partitions();
+          } catch (Exception exception) {
+            Thread.sleep(sleepTime);
+            processedSuccessfully = false;
+            LOGGER.warn("KafkaFuture failure. Will try once again.", exception);
+          }
+        } while (!processedSuccessfully);
+      }
+    }
+
     void maybeAddPartitions(int minPartitionNum) throws ExecutionException, InterruptedException {
       Collection<String> topicNames = _adminClient.listTopics().names().get();
       Map<String, KafkaFuture<TopicDescription>> kafkaFutureMap = _adminClient.describeTopics(topicNames).values();
```

Comment:
Why are we describing all topic partitions? Don't we just want the description of our singleton topic?